src/Experimental/ShortestPaths/ShortestPaths.jl

```diff
@@ -0,0 +1,212 @@
+    n == 0 && return Vector{T}()
+    p = Vector{T}(undef, n+1)
+    p[1] = src(s.path[1])
+    p[end] = dst(s.path[end])
+    for i in 2:n
+        p[i] = src(s.path[i])
+    end
+    return p
+end
+
+convert(::Type{AbstractPathState}, spr::AStarResults) = convert(AStarState, spr)
+convert(::Type{<:AStarResults}, s::AStarState) = AStarResults(s.path)
+
+convert(::Type{<:AStarState}, spr::AStarResults) = AStarState(spr.path)
+
+struct AStar{F<:Function} <: ShortestPathAlgorithm
+    heuristic::F
+end
+AStarResults(s::AStarState) = convert(AStarResults, s)
+AStarState(spr::AStarResults) = convert(AStarState, spr)
+
+AStar(T::Type{<:Real}=Float64) = AStar(n -> zero(T))
+
+
+##################
+## Bellman-Ford ##
+##################
+struct BellmanFord <: ShortestPathAlgorithm end
+struct BellmanFordResults{T<:Real, U<:Integer} <: ShortestPathResults
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+convert(::Type{AbstractPathState}, spr::BellmanFordResults) = convert(BellmanFordState, spr)
+convert(::Type{<:BellmanFordResults}, s::BellmanFordState) =
+    BellmanFordResults(s.parents, s.dists)
+
+convert(::Type{<:BellmanFordState}, spr::BellmanFordResults) =
+    BellmanFordState(spr.parents, spr.dists)
+
+
+
+BellmanFordResults(s::BellmanFordState) = convert(BellmanFordResults, s)
+BellmanFordState(spr::BellmanFordResults) = convert(BellmanFordState, spr)
+
+
+
+##################
+##Floyd-Warshall##
+##################
+
+struct FloydWarshall <: ShortestPathAlgorithm end
+struct FloydWarshallResults{T<:Real, U<:Integer} <: ShortestPathResults
+    parents::Matrix{U}
+    dists::Matrix{T}
+end
+
+convert(::Type{AbstractPathState}, spr::FloydWarshallResults) = convert(FloydWarshallState, spr)
+convert(::Type{<:FloydWarshallResults}, s::FloydWarshallState) =
+    FloydWarshallResults(s.parents, s.dists)
+convert(::Type{<:FloydWarshallState}, spr::FloydWarshallResults) = 
+    FloydWarshallState(spr.dists, spr.parents) # note - FWState is reversed from the others. Yuck.
+
+FloydWarshallResults(s::FloydWarshallState) = convert(FloydWarshallResults, s)
+FloydWarshallState(spr::FloydWarshallResults) = convert(FloydWarshallState, spr)
+
+##################
+##D'Esposo-Pape ##
+##################
+
+struct DEsopoPape <: ShortestPathAlgorithm end
+struct DEsopoPapeResults{T<:Real, U<:Integer} <: ShortestPathResults
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+convert(::Type{AbstractPathState}, spr::DEsopoPapeResults) = convert(DEsopoPapeState, spr)
+convert(::Type{<:DEsopoPapeResults}, s::DEsopoPapeState) =
+    DEsopoPapeResults(s.parents, s.dists)
+convert(::Type{<:DEsopoPapeState}, spr::DEsopoPapeResults) = 
+    DEsopoPapeState(spr.parents, spr.dists)
+
+DEsopoPapeResults(s::DEsopoPapeState) = convert(DEsopoPapeResults, s)
+DEsopoPapeState(spr::DEsopoPapeResults) = convert(DEsopoPapeState, spr)
+# if we don't pass in distances, use weights.
+shortest_paths(g::AbstractGraph, ss::Vector{T}, alg::ShortestPathAlgorithm) where {T<:Integer} =
```

Comment:
Do we really need a `Vector` in itself or `AbstractVector` as `ss`?

Reply:
`AbstractVector` should work.